narwhal/primary/src/core.rs

```diff
@@ -212,7 +212,19 @@ impl Core {
             .await;
         };
 
-        // Verify the vote.
+        // Verify the vote. Note that only the header digest is signed by the vote.
+        ensure!(
+            vote.digest == header.digest(),
+            DagError::UnexpectedVote(vote.digest)
+        );
```

Comment:
Since we are splitting the check here, shall we introduce a separate error for this? Otherwise I would be curious to see why this should be a separate check compared to how it was before

Reply:
I thought having different digest, vs having the same digest but unexpected origin / author should be treated differently. But probably this is not a big concern. Just moving the UnexpectedVote check to be the first check now.